includes/frontend.mjs

```diff
@@ -174,6 +177,75 @@ export function addDownloadTSVListener(downloadTSV) {
   });
 }
 
+function stringValue(value, other) {
+  return { userEnteredValue: { stringValue: value }, ...other };
+}
+
+function dateValue(date) {
+  // Courtesy of https://stackoverflow.com/a/64814390
+  const SheetDate = {
+    origin: Date.UTC(1899, 11, 30, 0, 0, 0, 0),
+    dayToMs: 24 * 60 * 60 * 1000,
+  };
+  const serial = (date.getTime() - SheetDate.origin) / SheetDate.dayToMs;
+  return numberValue(serial, { userEnteredFormat: { numberFormat: { type: 'DATE_TIME' } } });
+}
+
+function numberValue(value, other) {
+  return { userEnteredValue: { numberValue: value }, ...other };
+}
+
+export function getAppendGoogleRows(format="gapi") {
+  const exportTime = new Date();
+  const rows = [];
+  stockpiles.sort( (a, b) => a.lastModified - b.lastModified );
+  for (const stockpile of stockpiles) {
+    const stockpileTime = new Date(stockpile.lastModified);
+    //const stockpileID = Math.random().toString(36).replace(/^0\./, '');
+    const stockpileID = Math.floor(Math.random() * 1000000000000000);
+    for (const element of stockpile.contents) {
+      if (element.quantity == 0) {
+        continue;
+      }
```

Comment:
This is existing code, but I realize now this could cause an issue.  If the stockpile is empty, zero rows will be added and so no stockpile update will occur.

I think this is safe to take out.  I think I had it in there when there was math based on quantity below here.

Reply:
i believe taking this out would produce long rows of 0 for Shirts, bsups, gsups, whereas now missing items appear as emptystring. I rather opted to place a tombstone for empty stockpiles. ab116f4